### DISCUSSION TITLE
Real-Time Support?

### DISCUSSION THREAD

**Discussion by ajaybalachandran** [netflie/whatsapp-cloud-api#221] [General]
Does this PHP package support real-time chat functionality with customers using the WhatsApp Cloud API, allowing instant sending and receiving of messages?

**Comment by aindot** [comment]
If you mean by real-time that there will be open connection and exchange of data by that connection then no.

regardless, I'm already using this package to view clients chat and reply accordingly. upon deployment you will receive webhooks calls from Facebook, in which each call contain a message (or more than a message) the messages may be out of order, but the time-frame in which they will be delivered is very short.

**Reply by ajaybalachandran** [reply]
@aindot 
Thank you for your reply.
I am trying to build a CRM software in which I need to send and receive messages to and from customers and reply accordingly. The system should work in real-time, meaning there should be no delay in receiving customer messages, and no delay in receiving responses. Is this PHP library sufficient for this scenario?

**Reply by aindot** [reply]
Yes it is very much sufficient

**Comment by aalbarca** [comment]
Hi @ajaybalachandran this library is only a wrapper for the whatsapp cloud api calls.

The library is not a magic box so if your system is in real-time or not depends of you implementation, not this library.

When a customer send a message then Facebook will notify this to your webhook so we can say that is in "real-time" (regardless internet latency). If Facebook notify the message later or not this is not a concern of this library.

Best regards.
